modules/custom/az_digital_asset_library/az_digital_asset_library.module

```diff
@@ -6,51 +6,49 @@
  */
 
 use Drupal\Core\Form\FormStateInterface;
-use Drupal\Core\Url;
+use Drupal\Component\Serialization\Json;
 
 /**
- * Custom submit handler to redirect to /admin/content/media.
- */
-function az_digital_asset_library_form_submit(array $form, FormStateInterface $form_state) {
-  $url = Url::fromRoute('entity.media.collection');
-  $form_state->setRedirectUrl($url);
-}
-
-/**
- * Implements hook_FORM_ID_alter().
+ * Implements hook_preprocess_HOOK().
  *
- * Add custom submit handler to form.
- * Allow using remote media.
- * See
- * @link https://git.drupalcode.org/project/intelligencebank/-/blob/8.x-2.x/modules/ib_dam_media/src/Plugin/EntityBrowser/Widget/IbDamEbWidget.php#L249 @endlink.
+ * Hide default media type provided by IntelligenceBank from media add page.
+ *
+ * See @link https://github.com/az-digital/az_quickstart/issues/1741
  */
-function az_digital_asset_library_form_entity_browser_az_digital_asset_library_form_alter(&$form, FormStateInterface $form_state, $form_id) {
-  $form['#submit'][] = 'az_digital_asset_library_form_submit';
+function az_digital_asset_library_preprocess_entity_add_list(&$variables) {
+  if (isset($variables['bundles']['ib_dam_embed'])) {
+    unset($variables['bundles']['ib_dam_embed']);
+  }
 }
 
 /**
- * Implements hook_preprocess_HOOK().
+ * Implements hook_form_FORM_ID_alter().
  *
- * Hide default media type provided by IntelligenceBank from media select
- * dialog.
+ * Change label text for open browser button.
+ * Change dialog title option.
  *
- * See @link https://github.com/az-digital/az_quickstart/issues/1741
+ * See @link drupal.org/project/intelligencebank/issues/3353858
  */
-function az_digital_asset_library_preprocess_links__media_library_menu(&$variables) {
-  if (isset($variables['links']['media-library-menu-ib_dam_embed']['link'])) {
-    $variables['links']['media-library-menu-ib_dam_embed']['link']['#access'] = FALSE;
-  }
+function az_digital_asset_library_form_media_library_add_form_upload_alter(&$form, FormStateInterface $form_state, $form_id) {
+  $form['ib_dam_container']['asset_browser']['#title'] = t('Add Arizona Digital Asset Library Media');
+  $data_dialog_options = Json::decode($form['ib_dam_container']['asset_browser']['#attributes']['data-dialog-options']);
```

Comment:
@trackleft It looks like line 34 is resulting in some warnings in the [Probo log](https://7b5e2985-cedf-4a13-ab91-3a35916c835d--pr-2163.probo.build/admin/reports/dblog):

- Warning: Undefined array key "#attributes" in az_digital_asset_library_form_media_library_add_form_upload_alter() (line 34 of /src/az-quickstart-scaffolding/web/profiles/custom/az_quickstart/modules/custom/az_digital_asset_library/az_digital_asset_library.module)
- Warning: Trying to access array offset on value of type null in az_digital_asset_library_form_media_library_add_form_upload_alter() (line 34 of /src/az-quickstart-scaffolding/web/profiles/custom/az_quickstart/modules/custom/az_digital_asset_library/az_digital_asset_library.module)
- Deprecated function: json_decode(): Passing null to parameter #1 ($json) of type string is deprecated in Drupal\Component\Serialization\Json::decode() (line 26 of /src/az-quickstart-scaffolding/web/core/lib/Drupal/Component/Serialization/Json.php)

Would it be good to add some checking on the array value before running the Json::decode function?

Reply:
yes thanks